packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -95,20 +106,32 @@ export function getTextNodeAndOffset(node: Node, strOffset: number): NodeOffset
     domOffset = Math.max(0, domOffset - adjustment);
   }
 
+  console.log(
+    `Look through text nodes- strOffset: ${strOffset}, beginStrOffset: ${beginStrOffset}, domOffset: ${domOffset}`
+  );
+
   // In certain cases (i.e. text length > 65536 chars), the DOM will break
   // up text into multiple TextNodes. Loop through TextNodes, finding which
   // contains the `domOffset`
   const iterator = document.createNodeIterator(node, NodeFilter.SHOW_TEXT);
-  let textNode: Text, len: number;
+  let textNode: Text | null = null,
+    len: number,
+    prevNode: Text | null;
   do {
+    // @ts-ignore
+    console.log(`Iterate through text nodes: len: ${len}, domOffset: ${domOffset}`);
+    prevNode = textNode;
     textNode = iterator.nextNode() as Text;
   } while (textNode && (len = textNode.data.length) && domOffset > len && (domOffset -= len));
 
-  if (textNode === null) {
+  // We didn't find any text nodes
```

Comment:
Move this comment inside of the first `if` block; otherwise, it's ambiguous if it only applies to the first block or the entire if/else-if/else block.